### PR TITLE
Automate release process

### PR DESCRIPTION
**Pull request by vinicius73**
- NPM and GPR Publish action
- github-pages doc action
- prevent node14 test failure
- Bump version to 3.0.0

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: 📣 Publish package on NPM
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: 🔧​ Setup Github Registry
+        uses: actions/setup-node@v3
+        with:
+          registry-url: 'https://npm.pkg.github.com'
```
Comment (ChristophWurst) [inline .github/workflows/publish.yml:46]: is that where we publish?

Comment (vinicius73) [reply .github/workflows/publish.yml:46]: Now we publish only in NPM
But all major npm front-ends ([NPM](https://docs.npmjs.com/cli/v8/commands/npm-install), [Yarn](https://yarnpkg.com/cli/add), [pnpm](https://pnpm.io/cli/add#install-from-git-repository)) are able to use not only other registries, also github repos directly.

This movement is important to improve the security and resilience of the javascript ecosystem.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🔒​ Check actor permission level
+        uses: skjnldsv/check-actor-permission@v2
+        with:
+          require: admin
```
Comment (juliushaertl) [inline .github/workflows/publish.yml:14]: @skjnldsv Would this be enough as this is in line with https://github.com/nextcloud/nextcloud-vue/blob/master/.github/workflows/npm-publish.yml#L16-L21 or would we still need a separate repo on Nextcloud-releases for each package? (ref https://github.com/nextcloud/standards/issues/2#issuecomment-1102796417)

File: README.md [hunk 3]
```diff
@@ -31,14 +35,35 @@ unsubscribe('b', h)
...
+
+### Requirements
+
+- [Node 14 or higher](https://nodejs.org/en/download/)
+- [NPM 7 or higher](https://www.npmjs.com/package/npm)
+
+### Recommendations
+
+- [Node 16](https://nodejs.org/en/download/)
+- [NPM 8](https://www.npmjs.com/package/npm)
```
Comment (juliushaertl) [inline README.md:69]: We should add release documentation after https://github.com/nextcloud/nextcloud-event-bus/pull/490/files#r873962228 is clarified.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (skjnldsv) [reply .github/workflows/publish.yml:14]: Should be enough imho

File: .github/workflows/publish.yml (hunk 1 above)
Comment (skjnldsv) [reply .github/workflows/publish.yml:46]: Is it exclusive? Like does that means we'll only publish to the github registry instead of npm then?

Comment (juliushaertl) [reply .github/workflows/publish.yml:46]: No, npm is done in the previous step https://github.com/nextcloud/nextcloud-event-bus/pull/490/files/66382a3dfcf32553b70009e9a5ad5eb1f8e00511#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R38-R41

Comment (skjnldsv) [reply .github/workflows/publish.yml:46]: Perfect then

File: README.md (hunk 3 above)
Comment (vinicius73) [reply README.md:69]: I guess we can move forward.
https://github.com/nextcloud/standards/discussions/6

File: .github/workflows/node-test.yml [hunk 4]
```diff
@@ -19,6 +19,9 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: 'npm'
+      - name: setup npm@^7
```
Comment (juliushaertl) [inline .github/workflows/node-test.yml:22]: Can be dropped, we are using Node 16 now as a default: https://github.com/nextcloud/nextcloud-event-bus/commit/6f0429d39b926c3b3b648b9f45ecb7a22f32bb64

File: README.md [hunk 5]
```diff
@@ -31,14 +35,35 @@ unsubscribe('b', h)
...
+npm install
+
+npm run build
+npm run test
+```
+
+### Requirements
+
+- [Node 14 or higher](https://nodejs.org/en/download/)
+- [NPM 7 or higher](https://www.npmjs.com/package/npm)
```
Comment (juliushaertl) [inline README.md:64]:
```suggestion
- [Node 16 or higher](https://nodejs.org/en/download/)
- [NPM 7 or higher](https://www.npmjs.com/package/npm)
```

We are using Node 16 now as a default: https://github.com/nextcloud/nextcloud-event-bus/commit/6f0429d39b926c3b3b648b9f45ecb7a22f32bb64

**Review by juliushaertl — COMMENTED** [review]
Needs some adjustments after https://github.com/nextcloud/nextcloud-event-bus/commit/6f0429d39b926c3b3b648b9f45ecb7a22f32bb64 but otherwise LGTM 👍

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**